### PR TITLE
when clustername is empty remove extra dot in endpoints

### PR DESCRIPTION
**Pull request by jmichalek132**
Hi, in my case endpoints cannot contain clusterName. Without this change there is extra dot after svc. Would you be willing to accept this change?

Thank you.

### CODE REVIEW

**Comment by jmichalek132** [comment]
I gave up the linter wins.

File: go.mod [hunk 1]
```diff
@@ -4,17 +4,17 @@ go 1.14
...
 	github.com/onsi/ginkgo v1.11.0 // indirect
 	github.com/onsi/gomega v1.8.1 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.5.0
 	github.com/thanos-io/thanos v0.12.0
-	golang.org/x/crypto v0.0.0-20191206172530-e9b2fee46413 // indirect
 	k8s.io/api v0.0.0-20191115095533-47f6de673b26
 	k8s.io/apimachinery v0.0.0-20191115015347-3c7067801da2
 	k8s.io/client-go v12.0.0+incompatible
+	mvdan.cc/gofumpt v0.1.0 // indirect
```
Comment (bwplotka) [inline go.mod:17]: Delete this manually and/or run `go mod tidy` to make linter work (for a good reason -> This is not project dependency).

For tools I would recommend using https://github.com/bwplotka/bingo - it does not pollute project's go modules.

File: main.go [hunk 2]
```diff
@@ -517,14 +517,21 @@ func (c *controller) populate(hashrings []receive.HashringConfig, statefulsets m
 	for i, h := range hashrings {
 		if sts, exists := statefulsets[h.Hashring]; exists {
 			var endpoints []string
+
 			for i := 0; i < int(*sts.Spec.Replicas); i++ {
+				// if cluster Domain is empty string we don't want dot after svc
```
Comment (bwplotka) [inline main.go:522]: Can you make comments full sentence? 🤗 
(start with capital letter and end with dot)

File: main.go [hunk 3]
```diff
@@ -517,14 +517,21 @@ func (c *controller) populate(hashrings []receive.HashringConfig, statefulsets m
...
+				clusterDomain := ""
+				if len(c.options.clusterDomain) != 0 {
+					clusterDomain = fmt.Sprintf(".%s", c.options.clusterDomain)
+				}
```
Comment (bwplotka) [inline main.go:526]:
```suggestion
				clusterDomain := ""
				if c.options.clusterDomain != "" {
					clusterDomain = fmt.Sprintf(".%s", c.options.clusterDomain)
				}
```

(Does not matter much, just pattern 🤗 )

**Review by bwplotka — COMMENTED** [review]
LGTM, makes sense, thanks!

Ideally some short unit test would be amazing before merging! WDYT?

**Comment by bwplotka** [comment]
For lint `go mod tidy` should help

File: go.mod (hunk 1 above)
Comment (jmichalek132) [reply go.mod]: Thank you that fixed it.

File: main.go (hunk 2 above)
Comment (jmichalek132) [reply main.go]: Is this okay?

File: main.go (hunk 3 above)
Comment (jmichalek132) [reply main.go]: Changed :).

File: main_test.go [hunk 4]
```diff
@@ -233,6 +233,77 @@ func TestController(t *testing.T) {
 	}
 }
 
+func TestEndpointGenerationWithEmptyCluterName(t *testing.T) {
```
Comment (jmichalek132) [inline main_test.go:236]: Like this @bwplotka ?

am_i_doing_this_right

**Comment by jmichalek132** [comment]
Not sure why the test `TestController/OneHashringManyStatefulSets` failed after I added the new one. When I ran the test locally they succeed.

**Review by bwplotka — APPROVED** [review]
Thank you! LGTM!

**Pull request merged**